pynamodb/attributes.py

```diff
@@ -257,3 +261,223 @@ def deserialize(self, value):
         Takes a UTC datetime string and returns a datetime object
         """
         return parse(value, dayfirst=False).datetime
+
+
+class MapAttributeMeta(type):
+    def __init__(cls, name, bases, attrs):
+        setattr(cls, '_attributes', None)
+
+
+class MapAttribute(with_metaclass(MapAttributeMeta, Attribute)):
+    _attributes = None
+
+    def __init__(self, **attrs):
+
+        hash_key = attrs.get('hash_key', False)
```

Comment:
Is it actually valid to have a `Map` as a hash or range key? Can't find anything in the docs explicitly excluding it but it seems slightly weird (even if it's possible).